src/main/java/org/usfirst/frc4904/robot/RobotMap.java

```diff
@@ -68,8 +170,48 @@
     }
 
     public RobotMap() {
-        HumanInput.Driver.xbox = new CustomXbox(Port.HumanInput.xboxController);
-        HumanInput.Operator.joystick = new CustomJoystick(Port.HumanInput.joystick);
+        Component.pdp = new PDP();
+        Component.navx = new NavX(SerialPort.Port.kMXP);
+        Component.leftWheelEncoder = new CANCoder(Port.CAN.LEFT_WHEEL_ENCODER);
+        Component.rightWheelEncoder = new CANCoder(Port.CAN.RIGHT_WHEEL_ENCODER);
+        Component.canCoderConfiguration = new CANCoderConfiguration();
+        Component.leftWheelEncoder.configAllSettings(Component.canCoderConfiguration);
+        Component.rightWheelEncoder.configAllSettings(Component.canCoderConfiguration);
+        Component.leftWheelEncoder.configFeedbackCoefficient(RobotMap.Metrics.Chassis.METERS_PER_TICK, "meters",
+                SensorTimeBase.PerSecond);
+        Component.rightWheelEncoder.configFeedbackCoefficient(RobotMap.Metrics.Chassis.METERS_PER_TICK, "meters",
+                SensorTimeBase.PerSecond);
+
+        Component.leftWheelAccelerationCap = new EnableableModifier(new AccelerationCap(Component.pdp));
+        Component.leftWheelAccelerationCap.enable();
+        Component.rightWheelAccelerationCap = new EnableableModifier(new AccelerationCap(Component.pdp));
+        Component.rightWheelAccelerationCap.enable();
+
+        Component.leftDriveA = new Motor("leftDriveA", false, Component.leftWheelAccelerationCap,
+                new CANTalonFX(Port.CANMotor.LEFT_DRIVE_A));
+        Component.leftDriveB = new Motor("rightDriveA", false, Component.rightWheelAccelerationCap,
+                new CANTalonFX(Port.CANMotor.RIGHT_DRIVE_A));
+        Component.rightDriveA = new Motor("leftDriveB", false, Component.leftWheelAccelerationCap,
+                new CANTalonFX(Port.CANMotor.LEFT_DRIVE_B));
+        Component.rightDriveB = new Motor("rightDriveB", false, Component.rightWheelAccelerationCap,
+                new CANTalonFX(Port.CANMotor.RIGHT_DRIVE_B));
+
+        Component.turnPID = new CustomPIDController(PID.Turn.P, PID.Turn.I, PID.Turn.D, PID.Turn.F, (PIDSensor) Component.navx);
+        Component.shifter = new SolenoidShifters(Port.Pneumatics.SHIFTER.buildDoubleSolenoid());
+
+        Component.chassis = new TankDriveShifting(0.0, RobotMap.Component.leftDriveA, RobotMap.Component.leftDriveB,
```

Comment:
Note my earlier comment above—chassis in Components is currently just a TankDrive.